internal/eval/partial.go

```diff
@@ -0,0 +1,541 @@
+package eval
+
+import (
+	"errors"
+	"fmt"
+	"slices"
+
+	"github.com/cedar-policy/cedar-go/internal/ast"
+	"github.com/cedar-policy/cedar-go/types"
+)
+
+const variableEntityType = "__cedar::variable"
+
+func Variable(v types.String) types.Value {
+	return types.NewEntityUID(variableEntityType, v)
+}
```

Comment:
It's a little surprising to me that you didn't make a new type for variables and ignores that implements `IsScopeNode` and `Value`. Seems like a bit of a footgun to use an `EntityUID` to me because you have to be careful if you ever check for `v.(types.EntityUID)` to then also check if it's a real entityUID or an imposter.

There's probably some nuance I'm missing, but I'd love to talk it over with you.

Reply:
Let's talk that one over.